src/dexer.rs

```diff
@@ -1,10 +1,255 @@
-//! Builder for compiling Java source code into Android DEX bytecode.
-
-/// A builder for generating Android DEX bytecode by invoking `javac` and `d8` commands.
-///
-/// Documentation on `d8` options are based on
-/// <https://developer.android.com/tools/d8/>.
-#[derive(Clone, Debug, Default)]
-pub struct Dexer {
-    // TODO
-}
+//! Builder for compiling Java source code into Android DEX bytecode.
+
+use std::path::{Path, PathBuf};
+use std::ffi::{OsStr, OsString};
+use std::process::{Command, ExitStatus};
+use crate::env_paths::{self, PathExt};
+use crate::JavaRun;
+
+/// A builder for generating Android DEX bytecode by invoking `d8` commands.
+/// 
+/// Currently incremental building options are not provided here.
+/// 
+/// If you need to customize the `d8` command beyond what is provided here,
+/// you can use the [`Dexer::command()`] method to get a [`Command`]
+/// that can be further customized with additional arguments.
+/// 
+/// Documentation on `d8` options are based on
+/// <https://developer.android.com/tools/d8/>.
+#[derive(Clone, Debug, Default)]
+pub struct Dexer {
+    /// Override the default `JAVA_HOME` path.
+    /// Otherwise, the default path is found using the `JAVA_HOME` env var.
+    java_home: Option<PathBuf>,
+
+    /// Override the default `d8.jar` path.
+    /// Otherwise, the default path is found using [crate::android_d8_jar].
+    android_d8_jar_path: Option<PathBuf>,
+
+    /// Compile DEX bytecode without debug information. However, `d8` includes some information
+    /// that's used when generating stacktraces and logging exceptions.
+    release: bool,
+
+    /// Specify the minimum Android API level you want the output DEX files to support.
+    android_min_api: Option<u32>,
+
+    /// Disable Java 8 language features. Use this flag only if you don't intend to compile
+    /// Java bytecode that uses language features introduced in Java 8.
+    no_desugaring: bool,
+
+    /// Specify the path to the `android.jar` of your Android SDK.
```

Comment:
this one is missing the nice phrase that you included in the other fields: "Otherwise, the default path is found..."